app/main.py

```diff
@@ -1,34 +1,46 @@
+from dataclasses import dataclass
+
+
+@dataclass
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    row: int
+    column: int
+    is_alive: bool = True
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+            self, start: list[int], end: list[int], is_drowned: bool = False
+    ) -> None:
+        self.is_drowned = is_drowned
+        self.decks = [Deck(row, column)
+                      for row in range(start[0], end[0] + 1)
+                      for column in range(start[1], end[1] + 1)]
```

Comment:
In my opinion it would be better to separate the responsibility of creating a ship and creating its decks into different methods.
There are several ways to implement it. For example, you can make `decks` as an empty list in `__init__` and also add something like `self.create_ship()` which will call method for ship creation.

Reply:
thanks, good point